TACC/protx-dashboard
# Task/COOKS-323: refactor get selected geography name

## Overview: ##
Reduce the technical debt related to generating chart labels in dataUtils.

## Related Jira tickets: ##

* [COOKS-123](https://jira.tacc.utexas.edu/browse/COOKS-123)

## Summary of Changes: ##
- Converted PHR_MSA_County_Data from an array to an object.
- Defined a new function `getCountyName` to be used in `getSelectedGeographyName`
- Refactored `getSelectedGeographyName`
- Added new tests to dataUtils.test.js
- Misc changes: fix vite.config.ts, add /coverage to .gitignore, fix the maltreatment if block in MainChart

## Testing Steps: ##
1. Go to https://cep.test/protx/dash/demographics.
2. Make sure Counties is selected in the Area pulldown menu.
3. Click on the map to select a county and ensure the correct county name appears each time.

1. Go to https://cep.test/protx/dash/maltreatment.
2. Click on the map to select a county and ensure the correct county name appears each time.

1. Try out the new tests: from protx-client directory run `npm run test 'dataUtils.test.js'`

## UI Photos:

## Notes: ##
